### ISSUE TITLE
toi and toilim in `show()`

### ISSUE THREAD

**Issue by tensionhead** [esi-neuroscience/syncopy#240]
The return type of a call to `show` is different depending on the keyword:

```python
# out1 is an array as expected
out1 = my_data.show(toilim=[1, 2])
# out2 is a list with one entry, the (correct afaik) array
out2 = my_data.show(toi=np.linspace(1, 2, 5))
```

**Comment by tensionhead** [comment]
The same is true for the `foi/foilim` pair.. a `show(foi=[1,2,3])` returns a list with one entry

**Comment by tensionhead** [comment]
Similarly, for a selection outside of the data bounds with `toilim` a list holding one empty array is returned: 
```python
mock_up = np.arange(10).reshape(5, 2)
ad = spy.AnalogData([mock_up])
# ad.time gives [array([-1.,  0.,  1.,  2.,  3.])]
out = ad.show(toilim=[4, 5])  # this is a list holding the empty array
```
Repeating the same with `toi` will throw a misleading error message:
```python
out = ad.show(toi=[4, 5])
```
Throws:
`TypeError: Indexing elements must be in increasing order`

Here an array parsing as it is being done for `foi/foilim` would be needed!

**Comment by tensionhead** [comment]
So actually the same is true for `channel` selections..
```python
# out1 is a array
out1 = my_data.show(channel=0)
# out2 is a list
out2 = my_data.show(channel=[1,2])
```

**Comment by tensionhead** [comment]
The inconsistent return behaviour (list or not a list) got addressed in 7557e4a, the out of bounds errors should be addressed elsewhere!

**Comment by tensionhead** [comment]
Closed with #249
